code/stl/Complex.hpp

```diff
@@ -3,72 +3,101 @@
 template <typename T=float>
 class Complex_t {
 public:
-    Complex_t(T r, T i);
-    Complex_t();
-
-    T real() const {return m_r;};
-    T imaginary() const {return m_i;};
-
-    Complex_t operator+(const Complex_t& other) const;
-    Complex_t operator-(const Complex_t& other) const;
-    Complex_t operator*(const Complex_t& other) const;
-    Complex_t operator*(const T factor) const;
-    Complex_t operator/(const T dividend) const;
-    Complex_t& operator+=(const Complex_t& other);
-    bool operator<(const Complex_t& a) const;
+    Complex_t() = default;
+    Complex_t(T r, T i) : m_r(r), m_i(i) {}
+
+    T real() const { return m_r; }
+    T imaginary() const { return m_i; }
+
+    T norm() const {
+        return m_r * m_r + m_i * m_i;
```

Comment:
Here you're actually missing `sqrt`, aren't you?
This function would commonly be called norm2 or normSq.

Reply:
That's what I thought, but `std::norm` also returns the squared magnitude: https://en.cppreference.com/w/cpp/numeric/complex/norm

Reply:
I find this confusing, but maybe a small comment will clarify it?

Reply:
I renamed it to `norm_sqr()` now and added a new member function `norm()`, which does the sqrt.